codec-http2/src/main/java/io/netty/handler/codec/http2/Http2CodecUtil.java

```diff
@@ -104,12 +104,13 @@
     public static final int DEFAULT_WINDOW_SIZE = 65535;
     public static final short DEFAULT_PRIORITY_WEIGHT = 16;
     public static final int DEFAULT_HEADER_TABLE_SIZE = 4096;
+    public static final long INITIAL_SETTINGS_HEADER_LIST_SIZE = 8192;
```

Comment:
This is totally @exell-christopher's idea, but stealing it and putting it here: Just grabbing the setting from the initialHttp2 settings passed to the Http2Codec (if present) and using the max from there and defaulting to 8k if no custom settings were passed seems like it would nicely resolve this and maintain sane defaults for simpler cases.